### PR TITLE
Without netcdf

### PR DESCRIPTION
**Pull request by dschwoerer**
This adds support for `./configure --without-netcdf && make check`

I found #2003 during this, but other tests are now passing / being skipped.

Depends on #2000 to avoid merge conflicts.

### CODE REVIEW

File: tests/integrated/test-stopCheck-file/runtest [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #!/usr/bin/env python3
 
-#requires: not travis
+# requires: not travis
+# h5py doesn't load the data as expected
```
Comment (ZedThree) [inline tests/integrated/test-stopCheck-file/runtest:4]: Might be worth checking this is still the case

Comment (dschwoerer) [reply tests/integrated/test-stopCheck-file/runtest:4]: This is with python3-h5py-2.10.0
If the data-length is 1, it is returned as a number, rather, then an array of length 1.

Comment (ZedThree) [reply tests/integrated/test-stopCheck-file/runtest]: Fixed. It couldn't tell the difference between a time-dependent scalar with only one value, and an time-independent scalar

**Comment by ZedThree** [comment]
Tests are passing, just not getting reported correctly, again.

The change to tools/pylib/boututils/datafile.py in cc61817 should maybe go into master and/or the boutdata repo too

**Pull request merged**